addons/mail/static/src/messaging/entity/channel_partner/channel_partner.js

```diff
@@ -0,0 +1,72 @@
+odoo.define('mail.messaging.entity.ChannelPartner', function (require) {
```

Comment:
@seb-odoo : @alexkuhn  does not agree with naming the entity the same way it's named in python, please start a fight with him to decide which name I should use :p !

Reply:
`ChannelPartner` contains a lot more than the seen information.

This entity is the one that we need to properly compute a lot of thread features that current rely on formatter (and assume that the value of the current user is "the" value, which is a non ideal shortcut to make), such as channel rename per user, fold and minimize state, etc...

Reply:
I really think this is a mistake to have an entity that maps `channel_partner` table in python. This is an implementation detail in python, motivated by how it's stored in DB. This is good logic from monolith server PoV, but that's super wrong modeling in web-client and current partner PoV.

Some months before, I naïvely thought that our entities could simply be 1:1 map of python models. The reasons this is wrong is because entities should also model UI stuffs (unique entities that do not exist in python), should be aware of incompleteness (e.g. `message_unread_counter` cannot be deduced from all messages), and perceive uses differently (e.g. server consider typing notification as just generating longpolling notifications, while web-client has complex modeling with timers and enhancements on threads/partners/etc.).

So to sum up, `ChannelPartner` should not exist in entities, and instead should have several other meaningful entities like `ThreadPartnerSeenInfo`.